0.4.0

ad-ha opened the pull request:
- Initial deployment of badges revamping, as per details on #120 
- Major update of the code to merge all hard-coded text into consts

ccpk1 committed:
Update minor findings after 0.3.1 release

ad-ha committed:
Merge pull request #137 from ccpk1/0.3.1-Initial-Fixes

Update minor findings after 0.3.1 release

ad-ha committed:
Badges refactoring. Initial deployement

ad-ha committed:
Add DE translation

ad-ha committed:
Refactor const.py and flow_helpers imports

ad-ha committed:
Update constants and fix flow_helpers errors

ad-ha committed:
Revise constants and fix code typos

ad-ha committed:
Revise constants and general imports

ad-ha committed:
Revise const imports and remove deprecated sensors

ad-ha committed:
Revise const imports

ad-ha committed:
Revise const imports

ad-ha committed:
Revise const imports. Create function to remove deprecated sensors

ad-ha commented:
Major updates to all files for const imports management, deprecation of sensors and removal of deprecated entities.

Pending refactor of hard-coded text into constants on majority of files.

ad-ha committed:
Partial update consts from ConfigFlow and OptionsFlow

ad-ha committed:
Update ConfigFlow and OptionsFlow with test consts

ad-ha committed:
Update to constants

ad-ha committed:
Latest updates to Consts, Coordinator & OptionsFlow

ad-ha committed:
Update consts

ad-ha committed:
Finalized consts revision in OptionsFlow

ad-ha committed:
Fix import ordering

ad-ha committed:
Partial updates for consts in data management

ad-ha committed:
Fix issue with int AttributeError on removing entities. (#145)

ad-ha committed:
Update consts and fix failed overdue notifications

ad-ha committed:
Update consts

ad-ha committed:
Updated consts

ad-ha committed:
Update versioning

ad-ha committed:
Update to ConfigFlow and OptionsFlow consts

ad-ha committed:
Updates to new revamped badges

ad-ha committed:
Consts updates

ad-ha committed:
Update sensor to replace consts

ad-ha committed:
Adapt consts and register Bonuses Select

ad-ha committed:
Adapt consts for notification handling

ad-ha committed:
Fix consts definitions

ad-ha committed:
Adapt consts

ad-ha committed:
Add user definable Points Adjust Buttons via OptionsFlow

ad-ha committed:
Update button consts and UIDs. Fix deprecated buttons removal

ad-ha committed:
Fix Calendar consts. Make Period to Show user-definable in General Options in OptionsFlow

ad-ha committed:
Move duplicated functions into kc_helpers and adjust calls

ad-ha committed:
Updates to consts and minor changes to imports

ad-ha committed:
Refactor constants for clarity and consistency; update default values in coordinator and flow helpers

ad-ha committed:
Refactor imports and enhance badge migration logic for legacy support

ad-ha committed:
Add notification ID constant and enhance badge sensor documentation for clarity

ad-ha committed:
Fix chore streak updates and date management. Update rewards handling [ISSUE] Reward is 'charged' multiple times #142

ad-ha committed:
Fix typo in badge threshold constant names and enhance badge schema definitions

ad-ha committed:
Add assigned kids configuration for badges and enhance special occasions schema

ad-ha committed:
Remove redundant badge type from schemas

ad-ha committed:
Refactor cumulative badges handling and resets, enhance data and award modes.

ad-ha committed:
Update default badge award mode to 'award_none'

ad-ha committed:
Update translation files

ad-ha commented on custom_components/kidschores/translations/de.json:
@Blocksberg1988 do you mind having a look at the updated de.json that I changed to add new missing keys in preparation for 0.4.0?

Blocksberg1988 replied:
Klar kann ich es machen. ich ergänze dann nur mal die fehlenden Wörter und schicke bzw stelle es erneut ein

ad-ha replied:
Got it, thanks. I don't know German, sorry, had to get Google translator, but just to make it easier.

Blocksberg1988 replied:
No problem at all.
I'm not very good at English, so I also did the German translation with Google Translate.

And this was also my first time translating from English to German, at least for programming purposes regarding the language file.

But you never stop learning, and I enjoyed it. Why should I be stingy? 😂😂 I'm also happy for others, because others are happy for a lot here too! Thank you for this great extension.

ad-ha committed:
Fix translation key for threshold type in German, English, and Spanish files

ad-ha committed:
Update translation for special occasion badge in English

Blocksberg1988 reviewed (approved):
Ich habe die Datei überprüft und sieht gut aus. es wären ggf. nur kleine änderungen nötig. statt strafe soll es Sanktion heißen.

ad-ha committed:
Fix issue with Challenges Progress Calculations when a chore is defined on the Challenge

ad-ha committed:
Enhance LOGGER messages consistency across files.

ad-ha commented:
> Ich habe die Datei überprüft und sieht gut aus. es wären ggf. nur kleine änderungen nötig. statt strafe soll es Sanktion heißen.

@Blocksberg1988 feel free to propose any changes as you see fit. Thanks

Blocksberg1988 commented:
Bin die Datei noch mal durch gegangen! Das ist in Ordnung. Ich danke für die Mühe nochmal die Datei zu kontrollieren damit dies auch passt in der config

ad-ha committed:
Fix typo in constant name for assigned kids in const.py

ad-ha committed:
Update default value for badge maintenance rules to DEFAULT_ZERO in config and options flows

ad-ha committed:
Refactor points adjustment parsing to use '|' as a separator instead of one-per-line.

ad-ha committed:
Enhance LOGGER messages and improve consistency

ad-ha committed:
Refactor badge schema initialization to ensure rewards list is always populated and correct typo in translation for reset grace period description

ccpk1 committed:
Cumulative Badge Fixes

ad-ha committed:
Merge pull request #152 from ccpk1/20250402-Cumulative-Badge-Fixes

Cumulative Badge Fixes and Improvements

ad-ha committed:
Add missing translation keys for edit badges. Move error messages in options flow to const. Sort const alphabetically.

ccpk1 committed:
CFOF Changes for all badge types

ad-ha committed:
Merge pull request #154 from ccpk1/20250403-Badge-Fixes

CFOF Changes for all badge types

ad-ha committed:
Rebase removed consts before changes

ad-ha committed:
Flag migration tracking and versioning. Refactor periodic badge progress tracking for badges with chores.

ccpk1 committed:
Fix reset_type issues

ccpk1 committed:
Added missing de translation options for reset_type

ad-ha committed:
Merge pull request #155 from ccpk1/20250404---reset_type-fixes

Fix reset_type issues and handle new custom reset types

ccpk1 committed:
Add service to remove awarded badges

ad-ha committed:
Merge pull request #157 from ccpk1/20250405-Add-service-to-remove-awarded-badges

Add service to remove awarded badges from kids

ccpk1 committed:
Badge Fix-Enhance and _info variable normalization

ccpk1 committed:
Fix remove awarded badges to check before clearing fields

ccpk1 committed:
Merge pull request #2 from ccpk1:20250406-Part-2

Check for presense of field in remove awarded badges before clearing.

ccpk1 committed:
Refactor of Cumulative Badge Handling

ad-ha committed:
Merge pull request #159 from ccpk1/20250406-Badge-Enh-and-_info-variable-norm

Badge Fix-Enhance and _info variable normalization

ad-ha committed:
Merge pull request #160 from ccpk1/20250406-Changes-Part-3

Refactor of cumulative badge handling

ad-ha committed:
Fix data clean-up when resetting data with storage manager. #158

ccpk1 committed:
Add migration to badges_earned and New datetime helpers

ccpk1 committed:
Fix award count constant

ccpk1 committed:
Deprecate use of kid badges field

ccpk1 committed:
New datetime and schedule helpers - Cumulative badge maintentance implementated

ccpk1 committed:
Datetime helper improvements and cumulative edge case handling

ccpk1 committed:
Refactored chore scheduling - Enhanced datetime helpers - added DEFAULT_TIME_ZONE constant - Normalized time variable naming _iso _local and _utc

ccpk1 committed:
Refactored Badge Creation, Data Fields and Mgmt for all types. Update Options and Config Flow as well as en.json. Force point rounding to 1 digit. New global DEFAULT_TIME_ZONE constant.

ad-ha committed:
Merge pull request #161 from ccpk1/20250407-Development

20250411 Improvements in datetime and scheduling activities.  Rework and simplification of Badge management and data

ad-ha committed:
Fix translation file trailing spaces

ad-ha committed:
Fix badge reset condition to ensure frequency is not None before enabling reset

ccpk1 committed:
Badge Migration Updates and Scheduling Fixes

ccpk1 committed:
Constant comments

ad-ha committed:
Merge pull request #163 from ccpk1/20250412-Development

Badge Migration Updates and Scheduling Fixes

ccpk1 committed:
Fix bug in add badge - Fix cumulative badge threshold lookup - Implement assigned_to for cumulative.

ccpk1 committed:
Extend Kid Chore Data to support Badge Target Tracking

ccpk1 committed:
Enhanced Individual Chore Tracking: We now have detailed history in kid_chore_data:

Last claimed time
Last completed time
Last disapproved time
Last overdue time
Streak data
Period-based stats (daily/weekly/monthly)

ccpk1 committed:
Major badge logic refactor: unified badge progress tracking, new handler-based target evaluations including logic for assigned kids, multi chore selection, due dates, and overdue across all cases.  Implemented full featured recurring/reset scheduling.
Added daily, weekly, monthly, yearly stats tracking for chores and points
Enhanced badge schema and validation, including dynamic target type filtering
Added reusable helpers for chore/point progress and completion checks (Future use with Achievements and Challenges)

ccpk1 committed:
Badge Sensor Update

ccpk1 committed:
Refactor chore tracking: migrate to new kid_chore_data structure with period-based stats, implement kid_chore_stats aggregation for accurate daily/weekly/monthly/all-time metrics, and update helpers/sensors to use new data model.

ccpk1 committed:
Update Missing Stat

ccpk1 committed:
Rename and organize new functions

ccpk1 committed:
Badge target adjustments and form text changes.

ccpk1 committed:
Update Completed Chores Sensors

ccpk1 committed:
Add data tracking for points by period as well as introducing point stats.

ccpk1 committed:
Points data period cleanup functions and migration function.

ccpk1 committed:
Updates to period cleanup functions

ad-ha committed:
Merge pull request #166 from ccpk1/20250418---Development-2

Major Badge, Chore Data/Statistics, & Point Data/Statistics Enhancements: Architecture and Feature Expansion

ccpk1 committed:
Update Badge Migration Function

ccpk1 committed:
Update Badge Migration and Points Migration

ccpk1 committed:
Update All Time Point Stats

ccpk1 committed:
Update Chore Migration

ad-ha committed:
Merge pull request #167 from ccpk1/20250420---Development

Migration Updates

ccpk1 committed:
Add claimed to chore data / stats.  Standardize naming conventions in chores and points.  Migration fixes.

ad-ha committed:
Merge pull request #168 from ccpk1/20240421---Development

Add claimed to chore data / stats.  Standardize naming conventions in…

ccpk1 committed:
Refactor badge award/penalty logic, improve data migration, update constants, and enhance sensor attributes

ccpk1 committed:
Translation file adjustment for config flow

ad-ha committed:
Merge pull request #172 from ccpk1/20250423---Development

Badge Multi-Awards & Penalties, Data Migration Improvements, Constant Cleanup, and Sensor Enhancements

ccpk1 commented:
This is superseded by 0.5.0
